improver_tests/utilities/test_CalculateClimateAnomalies.py

```diff
@@ -0,0 +1,314 @@
+    )
+
+
+def test_calculate_standardised_anomalies_site_data(site_cubes):
+    """Test that the plugin calculates standardised anomalies correctly for site data."""
+    plugin = CalculateClimateAnomalies()
+    site_cube_diagnostic, _, _ = site_cubes
+    result = plugin.process(*site_cubes)
+    expected_anomalies = np.array([3.5], dtype=np.float32)
+    np.testing.assert_allclose(result.data, expected_anomalies, rtol=1e-5)
+    assert result.long_name == site_cube_diagnostic.name() + "_standard_anomaly"
+    assert result.units == "1"
+    assert "reference_epoch" in [coord.name() for coord in result.coords()]
+    assert (
+        "(CellMethod(method='anomaly', coord_names=('reference_epoch',), intervals=(), comments=())"
+        in str(result.cell_methods)
+    )
+
+
+def test_ignore_temporal_mismatch(diagnostic_cube, mean_cube, variance_cube):
+    """Test that the verify_time_coords_match() function handles requests to ignore temporal mismatch
+    between diagnostic cube and mean/variance cube.
+    """
+    plugin = CalculateClimateAnomalies(ignore_temporal_mismatch=True)
+    diagnostic_cube.coord("time").points = (
+        diagnostic_cube.coord("time").points + 10 * SECONDS_IN_HOUR
+    )  # Moves diagnostic bounds outside mean bounds
+    plugin.verify_time_coords_match(diagnostic_cube, mean_cube, variance_cube)
+
+
+## Testing the plugin's internal verification checks
+@pytest.mark.parametrize("error_to_check", ["mean_check", "variance_check"])
+def test_error_units_mismatch(
+    diagnostic_cube, mean_cube, variance_cube, error_to_check
+):
+    """Test that the plugin raises a ValueError if the units of the diagnostic and another cube mismatch"""
+    plugin = CalculateClimateAnomalies()
+    if error_to_check == "mean_check":
+        mean_cube.units = "C"  # The units should be K ordinarily
+        with pytest.raises(
+            ValueError,
+            match="The mean cube must have the same units as the diagnostic cube.",
+        ):
+            plugin.verify_units_match(diagnostic_cube, mean_cube, variance_cube=None)
+    else:
+        variance_cube.units = "K"  # The units should be K2 ordinarily
+        with pytest.raises(
+            ValueError, match="The variance cube must be the diagnostic cube squared."
+        ):
+            plugin.verify_units_match(diagnostic_cube, mean_cube, variance_cube)
+
+
+def test_error_spatial_coords_mismatch_gridded_data(
+    diagnostic_cube, mean_cube, variance_cube
+):
+    """Test that the plugin raises a ValueError if the spatial coordinates of the diagnostic cube and another cube mismatch"""
+    plugin = CalculateClimateAnomalies()
+    with pytest.raises(ValueError, match="The spatial coordinates must match."):
+        mean_cube.coord("latitude").points = mean_cube.coord("latitude").points + 20
+        mean_cube.coord("longitude").points = mean_cube.coord("longitude").points + 20
+        plugin.verify_spatial_coords_match(diagnostic_cube, mean_cube, variance_cube)
```

Comment:
```suggestion
def test_error_spatial_coords_mismatch_gridded_data(
    diagnostic_cube, mean_cube, variance_cube
):
    """Test that the plugin raises a ValueError if the spatial coordinates of the diagnostic cube and another cube mismatch"""
    mean_cube.coord("latitude").points = mean_cube.coord("latitude").points + 20
    mean_cube.coord("longitude").points = mean_cube.coord("longitude").points + 20
    plugin = CalculateClimateAnomalies()
    with pytest.raises(ValueError, match="The spatial coordinates must match."):
        plugin.verify_spatial_coords_match(diagnostic_cube, mean_cube, variance_cube)
```
We usually set up the test cubes outside of the `pytest.raises` command.